django/http/request.py

```diff
@@ -176,15 +177,29 @@ def build_absolute_uri(self, location=None):
             location = '//%s' % self.get_full_path()
         bits = urlsplit(location)
         if not (bits.scheme and bits.netloc):
-            current_uri = '{scheme}://{host}{path}'.format(scheme=self.scheme,
-                                                           host=self.get_host(),
-                                                           path=self.path)
-            # Join the constructed URL with the provided location, which will
-            # allow the provided ``location`` to apply query strings to the
-            # base path as well as override the host, if it begins with //
-            location = urljoin(current_uri, location)
+            # Handle the trivial case trivially. If the location is absolute
```

Comment:
The trivial case is now less trivial, with this change :) [I get the jist of what you're saying, though]

Reply:
Unfortunately, I spent probably half the time on this PR fixing the comments. If you have a better wording, please let me know. "Shortcut the trivial case" is the other one I came up with, but I liked the other one better.

Reply:
The wording *I* would use is probably `Handle the common case: If the location ...` as the suggestion is that a) it happens far more frequently b) its not the pathological case (being the one which calls `urljoin`, now)
I'm not going to endlessly bikeshed over it. It's basically fine and does its job as is.